### PR TITLE
marker action data edit inline

### PR DESCRIPTION
**Pull request by Fredaly**
action and data markers to edit inline

### CODE REVIEW

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp [hunk 1]
```diff
@@ -101,75 +101,36 @@ void USFSplineComponent::UpdateSpline()
...
-
-        LevelActorActionMarkers.Empty();
-
-        for ( const auto & data_marker : DataMarkers )
-        {
-            auto new_marker = create_marker( data_marker, NewObject< USFSplineMarkerObject_Data >( this ) );
-            Cast< USFSplineMarkerObject_Data >( new_marker.Object )->Data = data_marker.Data;
-            SplineMarkers.Emplace( new_marker );
-        }
+    Super::Serialize( archive );
```
Comment (TheEmidee) [inline Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: why did you remove the conversion functions?

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp [hunk 2]
```diff
@@ -101,75 +101,36 @@ void USFSplineComponent::UpdateSpline()
...
 
-        DataMarkers.Empty();
-    }
+    StaticActionMarkers.Empty();
+    LevelActorActionMarkers.Empty();
+    DataMarkers.Empty();
+}
 
-    Super::Serialize( archive );
+void USFSplineComponent::CreateMarkerFromDefault( const FSFSplineMarker & default_marker, FSFSplineMarker & new_marker )
```
Comment (TheEmidee) [inline Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: you can just return the constructed object

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp [hunk 3]
```diff
@@ -101,75 +101,36 @@ void USFSplineComponent::UpdateSpline()
...
+        if ( marker.Object != nullptr )
         {
-            auto new_marker = create_marker( marker, NewObject< USFSplineMarkerObject_Data >( this, marker.Object->GetClass() ) );
-            Cast< USFSplineMarkerObject_Data >( new_marker.Object )->Data = data_marker->Data;
-            SplineMarkers.Emplace( new_marker );
+            SplineMarkers.Emplace( FSFSplineMarker(
+                marker.Name,
+                marker.ItIsEnabled,
+                marker.Infos,
+                NewObject< USFSplineMarkerObject >( this, marker.Object->GetClass(), NAME_None, RF_NoFlags, marker.Object ) ) );
```
Comment (TheEmidee) [inline Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: why the last 3 arguments to NewObject?

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 1 above)
Comment (Fredaly) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: we cannot convert a `TSubClassOf `to `TObjectPtr ` instanced

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 3 above)
Comment (Fredaly) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: This is the way to create an Object with Default from an Template
image

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 2 above)
Comment (Fredaly) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: done

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 1 above)
Comment (TheEmidee) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: Then I propose you get rid of the legacy stuff and remove the 3 properties you clear below

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 1 above)
Comment (Fredaly) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: done

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp [hunk 4]
```diff
@@ -77,88 +77,35 @@ void USFSplineComponent::UpdateSpline()
...
-        if ( const auto * data_marker = Cast< USFSplineMarkerObject_Data >( marker.Object ) )
-        {
-            auto new_marker = create_marker( marker, NewObject< USFSplineMarkerObject_Data >( this, marker.Object->GetClass() ) );
-            Cast< USFSplineMarkerObject_Data >( new_marker.Object )->Data = data_marker->Data;
-            SplineMarkers.Emplace( new_marker );
+            SplineMarkers.Emplace( FSFSplineMarker(
+                marker.Name,
+                marker.ItIsEnabled,
+                marker.Infos,
+                NewObject< USFSplineMarkerObject >( this, marker.Object->GetClass(), NAME_None, RF_NoFlags, marker.Object ) ) );
```
Comment (Frickerson) [inline Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: you could use CreateMarkerFromDefault here?

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/SplineFollow/Private/Components/SFSplineComponent.cpp (hunk 4 above)
Comment (Fredaly) [reply Source/SplineFollow/Private/Components/SFSplineComponent.cpp]: Done

**Review by Frickerson — APPROVED** [review]

**Review by Frickerson — APPROVED** [review]

**Review by TheEmidee — APPROVED** [review]
